File: solutions/csharp/space-age/1/SpaceAge.cs
using System.Collections.Generic;

public enum Planet
{
    Mercury,
    Venus,
    Earth,
    Mars,
    Jupiter,
    Saturn,
    Uranus,
    Nepture
}

public class SpaceAge
{
    private const double EarthOrbitalPeriodInSeconds = 31_557_600;
    private static readonly Dictionary<Planet, double> OrbitalPeriodInEarthYearsByPlanets = new Dictionary<Planet, double>
    {
        {Planet.Mercury, 0.2408467},
        {Planet.Venus, 0.61519726},
        {Planet.Earth, 1},
        {Planet.Mars, 1.8808158},
        {Planet.Jupiter, 11.862615},
        {Planet.Saturn, 29.447498},
        {Planet.Uranus, 84.016846},
        {Planet.Nepture, 164.79132}
    };



    private readonly int Seconds;
    public SpaceAge(int seconds) =>
        Seconds = seconds;

    public double OnEarth() =>
        AgeOn(Planet.Earth);

    public double OnMercury() =>
        AgeOn(Planet.Mercury);

    public double OnVenus() =>
        AgeOn(Planet.Venus);

    public double OnMars() =>
        AgeOn(Planet.Mars);

    public double OnJupiter() =>
        AgeOn(Planet.Jupiter);

    public double OnSaturn() =>
        AgeOn(Planet.Saturn);

    public double OnUranus() =>
        AgeOn(Planet.Uranus);

    public double OnNeptune() =>
        AgeOn(Planet.Nepture);

    private double AgeOn(Planet planet) =>
        Seconds / EarthOrbitalPeriodInSeconds / OrbitalPeriodInEarthYearsByPlanets.GetValueOrDefault(planet, 1);
}

File: solutions/ocaml/space-age/1/space_age.ml
open Base

type planet =
  | Mercury
  | Venus
  | Earth
  | Mars
  | Jupiter
  | Saturn
  | Neptune
  | Uranus

let earth_year_in_seconds =
  31_557_600.0

let orbital_period_in_earth_years = function
  | Mercury -> 0.2408467
  | Venus   -> 0.61519726
  | Earth   -> 1.0
  | Mars    -> 1.8808158
  | Jupiter -> 11.862615
  | Saturn  -> 29.447498
  | Neptune -> 164.79132
  | Uranus  -> 84.016846

let seconds_to_earth_years seconds =
  Float.of_int seconds /. earth_year_in_seconds

let age_on planet seconds =
  let earth_years = seconds_to_earth_years seconds in
  let orbital_period = orbital_period_in_earth_years planet in
  earth_years /. orbital_period
